proteuslib/property_models/tests/test_NaCl_prop_pack.py

```diff
@@ -14,337 +14,139 @@
-        unscaled_constraint_list = list(unscaled_constraints_generator(m))
-        assert len(unscaled_constraint_list) == 0
-
-        # check if any variables are badly scaled
-        badly_scaled_var_list = list(badly_scaled_var_generator(m))
-        assert len(badly_scaled_var_list) == 0
-
-    @pytest.mark.unit
-    def test_dof(self, frame):
-        m = frame
-        assert (degrees_of_freedom(m) == 0)
-
-    @pytest.mark.component
-    def test_solve(self, frame):
-        m = frame
-
-        m.fs.stream.initialize()
-
-        results = solver.solve(m)
-
-        # Check for optimal solution
-        assert results.solver.termination_condition == \
-               TerminationCondition.optimal
-        assert results.solver.status == SolverStatus.ok
+class TestSeawaterProperty(PropertyTestHarness):
+    def configure(self):
+        self.prop_pack = props.NaClParameterBlock
+        self.param_args = {}
+        self.scaling_args = {('flow_mass_phase_comp', ('Liq', 'H2O')): 1,
+                             ('flow_mass_phase_comp', ('Liq', 'NaCl')): 1e2}
+        self.stateblock_statistics = {'number_variables': 20,
+                                      'number_total_constraints': 16,
+                                      'number_unused_variables': 1,  # pressure is unused
+                                      'default_degrees_of_freedom': 3}  # 4 state vars, but pressure is not active
+        self.default_solution = {('mass_frac_phase_comp', ('Liq', 'H2O')): 0.965,
+                                 ('mass_frac_phase_comp', ('Liq', 'NaCl')): 0.035,
+                                 ('dens_mass_phase', 'Liq'): 1021.5,
+                                 ('flow_vol_phase', 'Liq'): 9.790e-4,
+                                 ('conc_mass_phase_comp', ('Liq', 'H2O')): 985.7,
+                                 ('conc_mass_phase_comp', ('Liq', 'NaCl')): 35.75,
+                                 ('flow_mol_phase_comp', ('Liq', 'H2O')): 53.57,
+                                 ('flow_mol_phase_comp', ('Liq', 'NaCl')): 0.5989,
+                                 ('mole_frac_phase_comp', ('Liq', 'H2O')): 0.9889,
+                                 ('mole_frac_phase_comp', ('Liq', 'NaCl')): 1.106e-2,
+                                 ('molality_comp', 'NaCl'): 0.6206,
+                                 ('diffus_phase', 'Liq'): 1.472e-9,
+                                 ('visc_d_phase', 'Liq'): 1.055e-3,
+                                 ('osm_coeff', None): 0.9271,
+                                 ('pressure_osm', None): 2.853e6,
+                                 ('enth_mass_phase', 'Liq'): 9.974e4}
+
+@pytest.mark.component
+class TestNaClPropertySolution_1(PropertyRegressionTest):
+    def configure(self):
+        self.prop_pack = props.NaClParameterBlock
+        self.param_args = {}
 
-    @pytest.mark.component
-    def test_solution(self, frame):
-        m = frame
+        self.solver = 'ipopt'
```

Comment:
Since 'ipopt' is the default when using `get_solver()`, is it necessary to make this assignment?

Reply:
Yeah its not necessary, I guess I included it to show how to change the solver.